src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamProjectFileDependencyGroupConverter.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
+    /// A <see cref="Utf8JsonStreamReaderConverter{T}"/> to allow reading JSON into <see cref="ProjectFileDependencyGroup"/>
+    /// </summary>
+    internal class Utf8JsonStreamProjectFileDependencyGroupConverter : IUtf8JsonStreamReaderConverter<ProjectFileDependencyGroup>
+    {
+        public ProjectFileDependencyGroup Read(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException("Expected PropertyName, found " + reader.TokenType);
+            }
+
+            var frameworkName = reader.GetString();
+            reader.Read();
+            var dependencies = reader.ReadStringArrayAsIList(new List<string>());
```

Comment:
Is 'dependencies' ever empty? If so, please use `Array.Empty<string>` instead of `List<string>()` as the default.

Reply:
We should basically check all `new List` changes are consider whether Array.Empty is appropriate. 
https://github.com/NuGet/NuGet.Client/pull/5558#discussion_r1441142371.